### PR TITLE
fix(rest):new endpoint to delete all license information in admin tab

### PR DESCRIPTION
**Pull request by nikkuma7**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: #2064 
Create a new rest endpoint to delete all license information in admin tab

### How To Test?
1: http://localhost:8080/resource/api/licenses/delete
2: login with admin user and create token 
3: use token in postman. 
> How should these changes be tested by the reviewer?
> Have you implemented any additional tests?

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/LicenseController.java [hunk 1]
```diff
@@ -158,4 +158,12 @@ private HalResource<License> createHalLicense(License sw360License) {
         }
         return halLicense;
     }
+
+    @PreAuthorize("hasAuthority('WRITE')")
+    @RequestMapping(value = LICENSES_URL + "/delete", method = RequestMethod.DELETE)
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/LicenseController.java:163]: change the endpoint to /deleteAll.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java [hunk 2]
```diff
@@ -16,13 +16,19 @@
...
 import org.eclipse.sw360.datahandler.thrift.RequestStatus;
+import org.eclipse.sw360.datahandler.thrift.RequestSummary;
 import org.eclipse.sw360.datahandler.thrift.SW360Exception;
 import org.eclipse.sw360.datahandler.thrift.licenses.License;
 import org.eclipse.sw360.datahandler.thrift.licenses.Obligation;
 import org.eclipse.sw360.datahandler.thrift.licenses.LicenseService;
 import org.eclipse.sw360.datahandler.thrift.users.User;
+import org.eclipse.sw360.datahandler.thrift.users.UserGroup;
+import org.eclipse.sw360.datahandler.thrift.vendors.Vendor;
+import org.eclipse.sw360.datahandler.thrift.vendors.VendorService;
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java:31]: please remove unnecessary imports.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java [hunk 3]
```diff
@@ -75,6 +81,16 @@ public void deleteLicenseById(String licenseId, User user) throws TException {
         }
     }
 
+    public void deleteAllLicenseInfo(User user) throws TException {
+        LicenseService.Iface sw360LicenseClient = getThriftLicenseClient();
+        if(PermissionUtils.isUserAtLeast(UserGroup.ADMIN, user)){
+        RequestSummary deleteLicenseStatus = sw360LicenseClient.deleteAllLicenseInformation(user);
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java:87]: use proper spacing for if condition.

**Comment by rudra-superrr** [comment]
While testing one license is not getting deleted.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/LicenseController.java (hunk 1 above)
Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/LicenseController.java]: done.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java (hunk 2 above)
Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java]: done.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java (hunk 3 above)
Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java:84]: done.

**Comment by nikkuma7** [comment]
> While testing one license is not getting deleted.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java [hunk 4]
```diff
@@ -75,6 +78,15 @@ public void deleteLicenseById(String licenseId, User user) throws TException {
         }
     }
 
+    public void deleteAllLicenseInfo(User user) throws TException {
+        LicenseService.Iface sw360LicenseClient = getThriftLicenseClient();
+        if (PermissionUtils.isUserAtLeast(UserGroup.ADMIN, user)) {
+        RequestSummary deleteLicenseStatus = sw360LicenseClient.deleteAllLicenseInformation(user);
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java:84]: use proper indentation on line 84

Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java]: done.

**Comment by rudra-superrr** [comment]
code looks good.

**Pull request merged**